### ISSUE TITLE
Lock/race condition may occurs when adding children to a GroupProcedure with MutuallyExclusive condition

### ISSUE THREAD

**Issue by yageek** [ProcedureKit/ProcedureKit#792]
I created a [sample project](https://github.com/yageek/MutualExclusiveError) helping to reproduce the error. I'm using ProcedureKit 4.3.0.

In certain conditions, ProcedureKit seems to be locked if the children of a `GroupProcedure` are added after call to the initializer and one of the children contains a `MutuallyExclusive` condition.

Inside the sample code, the code related to the issue seems to be the following:

```
   init() {
        super.init(operations: [])
        name = "TopLevelGroup"

        let op = SimpleLowLevelProcedure()
        add(children: [op])
}
```
Removing the `MutuallyExclusive` condition solves the issue.
Adding the child inside the initializer instead of using `add` solves also the issue:
```
    init() {
     let op = SimpleLowLevelProcedure()
        super.init(operations: [op])
        name = "TopLevelGroup"
    }
```

**Comment by swiftlyfalling** [comment]
@yageek: I have a theory about what might be going on here, but it'll be a little bit before I have a chance to test it out. Have a fix in mind.

**Comment by danthorpe** [comment]
@swiftlyfalling I might have a bit of time to try out any theories if you want to discuss it. Had a play with @yageek's demo project yesterday with him - confirming the bug.

**Comment by swiftlyfalling** [comment]
@yageek, @danthorpe: The core problem is that condition evaluation (and several other steps, like mutual exclusivity locks) does not properly respect and handle the associated ProcedureQueue's `isSuspended` status.

The solution is complicated by the asynchronous nature and relationship of a number of tasks (condition evaluation, mutual exclusivity lock requests, suspending / un-suspending ProcedureQueues, starting Procedures, etc).

I should have a solid fix (and a bunch of additional tests) soon. This is a very nice edge-case catch!

**Comment by danthorpe** [comment]
Thanks for the overview of the underlying issue @swiftlyfalling!

**Comment by swiftlyfalling** [comment]
@yageek: A fix for this has now been merged into `development`. Please give it a try.

**Comment by yageek** [comment]
I just tested again with the sample app and I am not able to reproduce the issue. Great work 👍

**Comment by swiftlyfalling** [comment]
Fix now available in 4.4.0.
